### PR TITLE
Support duration_seconds in config when assuming roles

### PR DESCRIPTION
**Pull request by c42f**
Also some additional mocking in tests for interactive MFA tokens

### CODE REVIEW

**Comment by c42f** [comment]
@omus this might interest you. Thanks for #357, I've started using the MFA functionality from there already.

**Review by mattBrzezinski — APPROVED** [review]
bors try

**Comment by c42f** [comment]
Bump. Bors seems to have got stuck here?

File: test/AWSCredentials.jl [hunk 1]
```diff
@@ -90,7 +90,18 @@ end
...
+        server_time = now(UTC)
+        patches = [
+            Patches._assume_role_patch(
+                "AssumeRole";
+                access_key=access_key_id,
+                server_time=server_time,
+                token_code_ref=sent_token,
+            ),
+            Patches._getpass_patch(; secret=mfa_token),
+        ]
```
Comment (omus) [inline test/AWSCredentials.jl:104]: I'd make this a separate testset

**Comment by omus** [comment]
bors try

File: test/AWSCredentials.jl (hunk 1 above)
Comment (c42f) [reply test/AWSCredentials.jl:125]: I've split all the new tests out into their own test set now and left the "default profile" test as-is.

**Comment by c42f** [comment]
bors try

**Comment by mattBrzezinski** [comment]
bors try

**Comment by c42f** [comment]
I've added a commit to fix the test failure with web identities now (It seems these were relying on a zero session duration, so making the expiry time patchable with a more realistic session duration broke those tests.)

**Comment by mattBrzezinski** [comment]
bors try

**Comment by mattBrzezinski** [comment]
Awkwardly on mobile right now. I’ll add you as a collaborator in the morning so that you can do bors testing commands yourself!

**Comment by c42f** [comment]
Thanks!

**Comment by mattBrzezinski** [comment]
LGTM, I'll merge it in once #508 passes through and tag a new version shortly after.

**Pull request merged**